src/main/scala/io/github/pauljamescleary/petstore/infrastructure/repository/doobie/DoobieUserRepositoryInterpreter.scala

```diff
@@ -43,9 +48,13 @@ class DoobieUserRepositoryInterpreter[F[_]: Monad](val xa: Transactor[F])
 
   import UserSQL._
 
-  def put(user: User): F[User] =
+  def create(user: User): F[User] =
     insert(user).withUniqueGeneratedKeys[Long]("ID").map(id => user.copy(id = id.some)).transact(xa)
 
+  def update(user: User): F[Option[User]] = OptionT(user.id.flatTraverse(select(_).option)).flatMapF(
+    u => u.id.traverse(UserSQL.update(user, _).run.map(_ => user))
+  ).value.transact(xa)
```

Comment:
same comment here, favoring approachability if possible.

Reply:
Agree, this one is dumping way too many concepts into a single expression
.

Reply:
2598335 attempts to improve this